server/server.py

```diff
@@ -1,27 +1,52 @@
-import socket                                        
+import socket                                     
 from cryptography.fernet import Fernet
 import os
 import json
-import xml.etree.ElementTree as et
 import pickle as pk
+import xml.etree.ElementTree as et
+import xmltodict
 
 def json_format(file_name): #function for json
     with open(file_name, 'r') as pkfile:
         unpickled = json.loads(pkfile.read())
     with open(file_name, 'w') as unpkfile:
         unpkfile.write(str(unpickled))
 
+def json_dict(file_name): #function for json <class 'dict'>
+    with open(file_name, 'r') as dictfile:
+        unpickled = json.loads(dictfile.read())
+    return unpickled
+
 def binary_format(file_name): #function for binary
     with open(file_name,'rb') as pkfile:
         unpickled = pk.load(pkfile)
     with open(file_name, 'w') as unpkfile:
-        unpkfile.write(str(unpickled))
+        unpkfile.write(str())
+        
+def binary_dict(file_name): #function for binary <class 'dict'>
+    with open(file_name, 'rb') as pkfile:
+        unpickled = pk.load(pkfile, encoding='utf-8')
+        return unpickled
 
-def xml_format(file_name): # function for XML(temporary) (added to code)
+
+def xml_format(file_name): # function for XML
+    with open(file_name,'r') as pkfile: # added to code
+        unpickled = et.parse(pkfile) # added to code
+        root = unpickled.getroot()
+        write_file = et.tostring(root)
+        xml_decode = write_file.decode() 
+        xml_dict = xmltodict.parse(xml_decode) #change XML to dictionary
+    with open(file_name, 'w',) as unpkfile: # added to code
+        unpkfile.write(str(xml_dict)) # added to code
+
+def xml_format_dict(file_name): # function for XML <class 'dict'>
     with open(file_name,'r') as pkfile: # added to code
         unpickled = et.parse(pkfile) # added to code
-    with open(file_name, 'w') as unpkfile: # added to code
-        unpkfile.write(str(unpickled)) # added to code
+        root = unpickled.getroot()
+        write_file = et.tostring(root)
+        xml_decode = write_file.decode() # 
+        xml_dict = xmltodict.parse(xml_decode) #change XML to dictionary
```

Comment:
Replace xml_dict = xmltodict.parse(xml_decode)
with
xml_datadict = xmltodict.parse(xml_decode) #change XML to dictionary
xml_dict = dict(xml_datadict.pop('data'))

This results in the nested ordered dictionary being retrieved and converted back to dict.

Reply:
Changes have been added